### PR TITLE
feat(api): query events

### PR DESCRIPTION
**Pull request by hekike**
Add a `GET /api/v1/events` to retrieve most recent raw events.
Useful for debugging.

### CODE REVIEW

File: internal/streaming/clickhouse_connector/connector.go [hunk 1]
```diff
@@ -43,6 +45,23 @@ func NewClickhouseConnector(config ClickhouseConnectorConfig) (*ClickhouseConnec
...
+	subjects, err := c.queryEventsTable(ctx, namespace, params)
+	if err != nil {
+		if _, ok := err.(*models.NamespaceNotFoundError); ok {
+			return nil, err
+		}
+
+		return nil, fmt.Errorf("query events: %w", err)
+	}
+
+	return subjects, nil
```
Comment (tothandras) [inline internal/streaming/clickhouse_connector/connector.go:62]:
```suggestion
	return events, nil
```

File: internal/streaming/clickhouse_connector/connector.go [hunk 2]
```diff
@@ -43,6 +45,23 @@ func NewClickhouseConnector(config ClickhouseConnectorConfig) (*ClickhouseConnec
 	return connector, nil
 }
 
+func (c *ClickhouseConnector) QueryEvents(ctx context.Context, namespace string, params streaming.QueryEventsParams) ([]event.Event, error) {
+	if namespace == "" {
+		return nil, fmt.Errorf("namespace is required")
+	}
+
+	subjects, err := c.queryEventsTable(ctx, namespace, params)
```
Comment (tothandras) [inline internal/streaming/clickhouse_connector/connector.go:53]:
```suggestion
	events, err := c.queryEventsTable(ctx, namespace, params)
```
or just return here
```suggestion
	return c.queryEventsTable(ctx, namespace, params)
```

File: internal/streaming/clickhouse_connector/connector.go [hunk 3]
```diff
@@ -162,6 +181,62 @@ func (c *ClickhouseConnector) createEventsTable(ctx context.Context, namespace s
...
+
+	events := []event.Event{}
+
+	for rows.Next() {
+		var id string
+		var eventType string
+		var subject string
+		var source string
+		var time time.Time
+		var data string
```
Comment (tothandras) [inline internal/streaming/clickhouse_connector/connector.go:212]:
```suggestion
		var dataStr string
```

File: internal/streaming/clickhouse_connector/connector.go [hunk 4]
```diff
@@ -162,6 +181,62 @@ func (c *ClickhouseConnector) createEventsTable(ctx context.Context, namespace s
...
+
+		if err = rows.Scan(&id, &eventType, &subject, &source, &time, &data); err != nil {
+			return nil, err
+		}
+
+		// Parse data
+		var jsonData interface{}
+		err := json.Unmarshal([]byte(data), &jsonData)
```
Comment (tothandras) [inline internal/streaming/clickhouse_connector/connector.go:220]:
```suggestion
		var data interface{}
		err := json.Unmarshal([]byte(dataStr), &data)
```

File: internal/streaming/clickhouse_connector/connector.go [hunk 5]
```diff
@@ -162,6 +181,62 @@ func (c *ClickhouseConnector) createEventsTable(ctx context.Context, namespace s
...
+		}
+
+		event := event.New()
+		event.SetID(id)
+		event.SetType(eventType)
+		event.SetSubject(subject)
+		event.SetSource(source)
+		event.SetTime(time)
+		event.SetDataContentType("application/json")
+		event.SetData("application/json", jsonData)
```
Comment (tothandras) [inline internal/streaming/clickhouse_connector/connector.go:232]:
```suggestion
		event.SetData("application/json", data)
```

**Review by tothandras — CHANGES_REQUESTED** [review]

File: internal/server/router/router.go [hunk 6]
```diff
@@ -96,6 +96,39 @@ func (a *Router) IngestEvents(w http.ResponseWriter, r *http.Request, params api
...
+	logger := slog.With("operation", "queryEvents")
+
+	namespace := a.config.NamespaceManager.GetDefaultNamespace()
+	if params.NamespaceInput != nil {
+		namespace = *params.NamespaceInput
+	}
+
+	limit := 100
+	if params.Limit != nil {
+		limit = *params.Limit
```
Comment (sagikazarmark) [inline internal/server/router/router.go:109]: Does the OpenAPI validator verify that this is never larger than 100?

**Review by sagikazarmark — APPROVED** [review]
LGTM

File: internal/server/router/router.go (hunk 6 above)
Comment (hekike) [reply internal/server/router/router.go:109]: It doesn't, good catch. I added validation.

**Pull request merged**